src/App.test.js

```diff
@@ -1,24 +1,133 @@
 import React from "react";
 import ReactDOM from "react-dom";
 import { App } from "./App";
+import { BlogPost } from "./BlogPost";
 import { render } from "react-testing-library";
+import { AllPosts } from "./ALlPosts";
 
 it("renders without crashing", () => {
   const div = document.createElement("div");
   ReactDOM.render(<App />, div);
   ReactDOM.unmountComponentAtNode(div);
 });
 
-it("renders a message", () => {
+it("renders entire header", () => {
   const { getByText } = render(<App />);
-  const message = getByText("Hello there");
-  expect(message).toBeInTheDocument();
-  expect(message).toHaveClass("message");
-  expect(message).toBeVisible();
+  const title = getByText("A Seattle Coder's Learning Blog");
+  expect(title).toBeInTheDocument();
+  expect(title).toHaveClass("title");
+  expect(title).toBeVisible();
+});
+
+it("renders main title and subtitle", () => {
+  const { getByText } = render(<App />);
+  const subtitle = getByText(
+    "A Woman in Tech, Breaking Stereotypes and Broadening Inclusion."
+  );
+  expect(subtitle).toBeInTheDocument();
+  expect(subtitle).toHaveClass("header-subtitle");
+  expect(subtitle).toBeVisible();
+});
+
+it("renders main nav and nav links", () => {
+  const { getByTitle } = render(<App />);
+  const getMainNav = getByTitle("main-nav-bar");
+  expect(getMainNav).toBeInTheDocument();
+  expect(getMainNav).toHaveClass("main-nav");
+  expect(getMainNav).toBeVisible();
+
+  const getNavLinks = getByTitle("main-nav-link");
+  expect(getNavLinks).toBeInTheDocument();
+  expect(getNavLinks).toHaveClass("nav-link");
+  expect(getNavLinks).toBeVisible();
+});
+
+it("renders a post", () => {
+  const post = {
+    id: 0,
+    title: "Mary had a little lamb, little lamb, little lamb.",
+    author: "Charlotte Mickey Mouse",
+    createdOn: "2019-01-02",
+    body: "No worries.",
+    tags: ["trees", "rocks"]
+  };
 
-  // expe
+  const authorRegex = new RegExp(post.author);
+
+  const { getByText } = render(<BlogPost post={post} />);
+  expect(getByText(post.title)).toBeInTheDocument();
+  expect(getByText(post.title)).toHaveClass("post-title");
+
+  expect(getByText(authorRegex)).toBeInTheDocument();
+  expect(getByText(authorRegex)).toHaveClass("post-author");
+
+  const newDate = new RegExp("Publish Date:");
+
+  expect(getByText(newDate)).toBeInTheDocument();
+  expect(getByText(newDate)).toHaveClass("date-created");
+
+  expect(getByText(post.body)).toBeInTheDocument();
+  expect(getByText(post.body)).toHaveClass("post-body");
+
+  const tagsString = new RegExp("Key Words:");
```

Comment:
This is only testing that the "key words" span is rendering. We haven't actually tested that the tags themselves are being rendered, so let's add more testing here to show that the "trees" and "rocks" tags are also being rendered. This case is different from the `createdOn` test because `createdOn` all renders in a single HTML element, whereas the tags each have their own element, so we could test the whole `createdOn` by matching just "publish date", but here our matcher is not getting all the elements that comprise the tags section